### DISCUSSION TITLE
upscaler support

### DISCUSSION THREAD

**Discussion by NikitaPZ** [Amblyopius/Stable-Diffusion-ONNX-FP16#6] [Ideas]
I think it would be super convenient to have checkboxes for ESRGAN and GFGPAN in the UI. right now i use both of them without problems on AMD externally, so seems like it would be possible to include those

https://github.com/TencentARC/GFPGANESRGAN
https://github.com/xinntao/Real-ESRGAN

**Comment by Amblyopius** [comment]
I personally also recommend https://github.com/xinntao/Real-ESRGAN to anyone who asks. Though then just the .exe version implemented with NCNN.

Integrated support in Python is slightly less convenient for those hence the current options are:
a) Microsoft fixes torch-directml and we integrate ESRGAN and GFPGAN like that (currently broken)
b) We convert to ONNX and run it like that. The latter is definitely possible, just takes some time to build an ONNX pipeline for it.

Final decision on direction will probably be soon (once the dust settles after the changes I plan for when ORT 1.14 comes out)

**Reply by ssube** [reply]
I've been able to convert both SD upscaling and Real ESRGAN to ONNX and get them working, with some fairly thin mocks to make the Real ESRGAN code think it was still dealing with Torch tensors: https://github.com/ssube/onnx-web/blob/main/api/onnx_web/onnx/onnx_net.py (SD took a whole new pipeline)

It seems to work correctly and does not take a lot of VRAM, `1950MiB / 24576MiB`.
